src/bun.js/api/bun/x509.zig

```diff
@@ -0,0 +1,553 @@
+        // We intentionally ignore the value of X509_NAME_ENTRY_set because the
+        // representation as an object does not allow grouping entries into sets
+        // anyway, and multi-value RDNs are rare, i.e., the vast majority of
+        // Relative Distinguished Names contains a single type-value pair only.
+        const type_ = BoringSSL.X509_NAME_ENTRY_get_object(entry);
+
+        // If BoringSSL knows the type, use the short name of the type as the key, and
+        // the numeric representation of the type's OID otherwise.
+        const type_nid = BoringSSL.OBJ_obj2nid(type_);
+        var type_buf: [80]u8 = undefined;
+        var name_slice: []const u8 = undefined;
+        if (type_nid != BoringSSL.NID_undef) {
+            const type_str = BoringSSL.OBJ_nid2sn(type_nid);
+            if (type_str == null) {
+                continue;
+            }
+            name_slice = type_str[0..bun.len(type_str)];
+        } else {
+            const length = BoringSSL.OBJ_obj2txt(&type_buf, @sizeOf(@TypeOf(type_buf)), type_, 1);
+            if (length <= 0) {
+                continue;
+            }
+            name_slice = type_buf[0..@intCast(usize, length)];
+        }
+
+        const value_data = BoringSSL.X509_NAME_ENTRY_get_data(entry);
+
+        var value_str: [*c]u8 = undefined;
+        const value_str_len = BoringSSL.ASN1_STRING_to_UTF8(&value_str, value_data);
+        if (value_str_len < 0) {
+            continue;
+        }
+        const value_slice = value_str[0..@intCast(usize, value_str_len)];
+        defer BoringSSL.OPENSSL_free(value_str);
+        // For backward compatibility, we only create arrays if multiple values
+        // exist for the same key. That is not great but there is not much we can
+        // change here without breaking things. Note that this creates nested data
+        // structures, yet still does not allow representing Distinguished Names
+        // accurately.
+        if (result.getTruthy(globalObject, name_slice)) |value| {
+            if (value.jsType().isArray()) {
+                value.push(globalObject, JSC.ZigString.fromUTF8(value_slice).toValueGC(globalObject));
+            } else {
+                const prop_name = JSC.ZigString.fromUTF8(name_slice);
+                const array = JSValue.createEmptyArray(globalObject, 2);
+                array.putIndex(globalObject, 0, value);
+                array.putIndex(globalObject, 1, JSC.ZigString.fromUTF8(value_slice).toValueGC(globalObject));
+                result.put(globalObject, &prop_name, array);
+            }
+        } else {
+            const prop_name = JSC.ZigString.fromUTF8(name_slice);
+            result.put(globalObject, &prop_name, JSC.ZigString.fromUTF8(value_slice).toValueGC(globalObject));
+        }
+    }
+    return result;
+}
+
+inline fn isSafeAltName(name: [*c]const u8, length: usize, utf8: bool) bool {
```

Comment:
why is this a c pointer and not a slice?

Reply:
basically because I translated some C++ -> Zig, i will refactor this one.

Reply:
fix on https://github.com/oven-sh/bun/commit/927f54269bdf489d21be4a8a6b89b448fdf48184